src/irmin-test/store.ml

```diff
@@ -1413,6 +1417,361 @@ module Make (S : Generic_key) = struct
+          "inside: find bar/age in proof" (Some "2") i;
+        let* i = S.Tree.find t [ "bar"; "version" ] in
+        Alcotest.(check (option string))
+          "inside: find bar/version in proof" (Some "3") i;
+        let* i = S.Tree.find t [ "hello"; "there" ] in
+        Alcotest.(check (option string))
+          "inside: do not find hello/there in proof" None i;
+        let+ () =
+          Lwt.catch
+            (fun () ->
+              let+ _ = S.Tree.find t [ "foo"; "version" ] in
+              Alcotest.fail "inside: should have raise: pruned_hash exn")
+            (function
+              | S.Tree.Pruned_hash _ | B.Node.Val.Dangling_hash _ ->
+                  Lwt.return ()
+              | e -> Lwt.fail e)
+        in
+        ()
+      in
+
+      let check_proof_f1 p =
+        let t = S.Tree.Proof.to_tree p in
+        let+ i = S.Tree.find t [ "foo"; "version" ] in
+        Alcotest.(check (option string))
+          "outside: find foo/version" (Some "1") i
+      in
+
+      let f0 t0 =
+        let* t1 = S.Tree.update t0 [ "foo"; "age" ] increment in
+        let* t2 = S.Tree.update t1 [ "bar"; "age" ] increment in
+        let* t3 = S.Tree.get_tree t2 [ "bar" ] in
+        let* t4 = S.Tree.add_tree t2 [ "hello"; "there" ] t3 in
+        let* v = S.Tree.get t4 [ "hello"; "there"; "version" ] in
+        Alcotest.(check string) "hello/there/version" "3" v;
+        let t = S.Tree.empty () in
+        let* t5 = S.Tree.add_tree t [ "dir1"; "dir2" ] t4 in
+        let* v = S.Tree.get t5 [ "dir1"; "dir2"; "bar"; "age" ] in
+        Alcotest.(check string) "dir1/dir2/bar/age" "3" v;
+        let* t = S.Tree.remove t4 [ "bar" ] in
+
+        (* Trigger certain paths in [S.Tree] during "verify" *)
+        let portable =
+          (* During "verify" [portable] is [Pruned] with [portable] in env *)
+          t0
+        in
+        let portable_dirty = t in
+        let trigger_node_to_map t =
+          S.Tree.fold ~depth:(`Eq 1) ~order:`Sorted ~force:`True t ()
+        in
+        let* () = trigger_node_to_map portable in
+        let* () = trigger_node_to_map portable_dirty in
+        let trigger_node_length t =
+          let+ (_ : int) = S.Tree.length t [] in
+          ()
+        in
+        let* () = trigger_node_length portable in
+        let* () = trigger_node_length portable_dirty in
+        let trigger_node_fold_undefined t =
+          S.Tree.fold ~depth:(`Eq 1) ~order:`Undefined ~force:`True t ()
+        in
+        let* () = trigger_node_fold_undefined portable in
+        let* () = trigger_node_fold_undefined portable_dirty in
+        let (_ : bool) = S.Tree.is_empty portable in
+        let trigger_node_to_backend_portable t =
+          match S.Tree.destruct t with
+          | `Contents _ -> assert false
+          | `Node n ->
+              let+ _ = S.to_backend_portable_node n in
+              ()
+        in
+        let* () = trigger_node_to_backend_portable portable_dirty in
```

Comment:
These tests are new. They cover all the `Portable*` cases when using `Scan` in tree